src/backend/optimizer/plan/createplan.c

```diff
@@ -4409,10 +4409,24 @@ create_ctescan_plan(PlannerInfo *root, Path *best_path,
 			RelOptInfo *sub_final_rel;
 			GangType	saved_gangType = root->curSlice->gangType;
 
+			/*
+			 * Temporarily clear direct dispatch info to obtain a decision only for the subplan,
+			 * without the outer plan affecting it.
+			 */
+			DirectDispatchInfo savedDirectDispatch = root->curSlice->directDispatch;
```

Comment:
Do you really need child only direct dispatch info here? The executor expects only the same set of hosts for producer/consumers. This set depends on the slice that executes this share scan. Why it's better than merging of info-s for its driver slice?

Reply:
Fixed. Slice's DDI can indeed be safely copied since we're merging them all anyway.